openstack/02b-MAChigh/neighbors.c

```diff
@@ -681,6 +682,14 @@ void registerNewNeighbor(open_addr_t *address,
                 if (rssi < GOODNEIGHBORMINRSSI) {
                     break;
                 }
+                open_addr_t addr_16;
+                addr_16.type = ADDR_16B;
+                packetfunctions_mac64bToMac16b(address, &addr_16);
+                uint16_t short_addr = ((addr_16.addr_16b[0] << 8) & 0xff00) |

```

Comment:
I was thinking here whether we did this sort of conversion from `ADDR_16B` type to a `uint16` anywhere else in the code but did not find a similar use case. a short address is logged for `ERR_NO_NEXTHOP` but that implementation requires two arguments which is not applicable to your case since you want to log both the short address and the RSSI. With that in mind, I am ok with this approach.